compositor_render/src/scene/view_component.rs

```diff
@@ -0,0 +1,188 @@
+            .into_iter()
+            .flatten()
+            .collect();
+
+        Ok(BaseNode::Layout {
+            root: LayoutComponent::View(self.clone()),
+            children,
+        })
+    }
+
+    pub(super) fn children(&self) -> Vec<&Component> {
+        self.children.iter().collect()
+    }
+
+    pub(super) fn children_mut(&mut self) -> Vec<&mut Component> {
+        self.children.iter_mut().collect()
+    }
+
+    pub(super) fn layout(&self, size: Resolution) -> Vec<NestedLayout> {
+        let dynamic_child_size = self.dynamic_size_for_static_children(size);
+
+        // offset along x or y direction (depends on self.direction) where next
+        // child component should be placed
+        let mut static_offset = 0;
+
+        let create_static_layout = match self.direction {
+            ViewChildrenDirection::Row => Self::create_static_layout_row,
+            ViewChildrenDirection::Column => Self::create_static_layout_column,
+        };
+
+        self.children
+            .iter()
+            .map(|child| {
+                if Self::is_static_child(child) {
+                    let layout =
+                        create_static_layout(child, static_offset, dynamic_child_size, size);
+                    match self.direction {
+                        ViewChildrenDirection::Row => static_offset += layout.width as usize,
+                        ViewChildrenDirection::Column => static_offset += layout.height as usize,
+                    };
+                    match child {
+                        Component::Layout(layout_component) => {
+                            let children_layouts = layout_component.layout(Resolution {
+                                width: layout.width as usize,
+                                height: layout.height as usize,
+                            });
+                            NestedLayout {
+                                layout: Layout {
+                                    content: LayoutContent::Color(
+                                        layout_component
+                                            .background_color()
+                                            .unwrap_or(RGBAColor(0, 0, 0, 0)),
+                                    ),
+                                    ..layout
+                                },
+                                children: children_layouts,
+                            }
+                        }
+                        _ => {
+                            NestedLayout {
+                                layout: layout.clone(),
+                                children: vec![NestedLayout {
+                                    layout: Layout {
+                                        top: 0.0,
+                                        left: 0.0,
+                                        content: LayoutContent::ChildNode(0), // TODO: this will be recalculated latter
```

Comment:
This is probably the most hacky part of the entire approach. I'm relaying here on the fact that when traversing the component tree I will visit all children components that represent nodes in the same order as when traversing the Node tree.

In particular, this means that this approach will only work if all children nodes will be always used in the layout. If we introduce component that has children that only conditionally show up in layout this will stop working.